Source/Diagnostics/ReducedDiags/ParticleNumber.cpp

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2019-2020 Neil Zaim, Yinjian Zhao
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "ParticleNumber.H"
+#include "WarpX.H"
+
+// constructor
+ParticleNumber::ParticleNumber (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+    // get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    // get MultiParticleContainer class object
+    auto & mypc = warpx.GetPartContainer();
+
+    // get number of species (int)
+    auto nSpecies = mypc.nSpecies();
+
+    // resize data array to nSpecies+1
+    // (number of particles of each species + total number of particles)
+    m_data.resize(nSpecies+1, amrex::Real(0.));
+
+    // get species names (std::vector<std::string>)
+    auto species_names = mypc.GetSpeciesNames();
+
+    if (amrex::ParallelDescriptor::IOProcessor())
+    {
+        if ( m_IsNotRestart )
+        {
+            // open file
+            std::ofstream ofs;
+            ofs.open(m_path + m_rd_name + "." + m_extension,
+                std::ofstream::out | std::ofstream::app);
+            // write header row
+            ofs << "#";
+            ofs << "[1]step()";
+            ofs << m_sep;
+            ofs << "[2]time(s)";
+            ofs << m_sep;
+            ofs << "[3]total";
```

Comment:
I just remember we had a purpose for adding `()` even for those quantities without a unit.
Because, we had a `read_reduced_diags` function in `WarpX/Tools/PostProcessing/read_raw_data.py`, which requires `()` to get the names of the quantities.
Could you please add those?
E.g., change `ofs << "[3]total";` to `ofs << "[3]total()";`.

Reply:
Thanks, I didn't know that. I've added the parentheses in the last commit.